<?xml version="1.0" encoding="utf-8"?>
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL 2.2] Single Sign-On - ABAP" id="BL22_SSO-A" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <!-- SSO-A: Single Sign-On â€“ ABAP -->
  <configstore name="ABAP_INSTANCE_PAHI">
    <checkitem desc="[p2-STANDARD] Generate ticket only via https" id="SSO-A_a">
      <compliant>          NAME = 'login/ticket_only_by_https'  and  VALUE = '1'     </compliant>
      <complianttext/>
      <noncompliant>    NAME = 'login/ticket_only_by_https'  and VALUE != '1'    </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Ticket only back to creating host" id="SSO-A_b">
      <compliant>          NAME = 'login/ticket_only_to_host'  and VALUE = '1'       </compliant>
      <complianttext/>
      <noncompliant>    NAME = 'login/ticket_only_to_host'  and VALUE != '1'     </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Set HTTPonly-Flag for Cookies" id="SSO-A_c">
      <compliant>          NAME = 'icf/set_HTTPonly_flag_on_cookies'  and not ( VALUE = '1'or VALUE = '3')   </compliant>
      <complianttext/>
      <noncompliant>    NAME = 'icf/set_HTTPonly_flag_on_cookies'  and ( VALUE = '1'or VALUE = '3')         </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
</targetsystem>