<?xml version="1.0" encoding="utf-16"?>
<!--
This FRUN CSA policy contains rules derived from SAP Security Baseline Template Version 3
Available at: http://support.sap.com/sos

Target system BL_S-4 / Secure Configuration of Java Systems

Version: 001
Date:    Feb 02 2018
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="Secure Configuration of Java Systems" id="BL_S-4" multisql="Yes" xsi:schemaLocation="csa_policy.xsd">
 <configstore name="http">
  <checkitem desc="Disable Server Header" id="BL_S-4.01">
   <compliant>NAME = 'UseServerHeader' and VALUE = 'false'</compliant>
   <complianttext/>
   <noncompliant>NAME = 'UseServerHeader' and not ( VALUE = 'false'  )</noncompliant>
   <noncomplianttext/>
  </checkitem>
 </configstore>
 <configstore name="servlet_jsp">
  <checkitem desc="Disable Invoker Servlet" id="BL_S-4.02">
   <compliant>NAME = 'EnableInvokerServletGlobally' and VALUE = 'false'</compliant>
   <complianttext/>
   <noncompliant>NAME = 'EnableInvokerServletGlobally' and not ( VALUE = 'false'  )</noncompliant>
   <noncomplianttext/>
  </checkitem>
 </configstore>
 <configstore name="xmlhardener_srv">
  <checkitem desc="Enable XML Hardener" id="BL_S-4.03">
   <compliant>NAME = 'enable.xml.hardener' and VALUE = 'true'</compliant>
   <complianttext/>
   <noncompliant>NAME = 'enable.xml.hardener' and not ( VALUE = 'true'  )</noncompliant>
   <noncomplianttext/>
  </checkitem>
 </configstore>
</targetsystem>

