<?xml version="1.0" encoding="utf-16"?>
<!--
This FRUN CSA policy contains rules derived from SAP Security Baseline Template Version 3
Available at: http://support.sap.com/sos

Target system BL_S-3 / Modification Protection for Production Systems

Version: 001
Date:    Feb 02 2018
-->
<targetsystem desc="Modification Protection for Production Systems" id="BL_S-3" multisql="Yes">
  <configstore name="CLIENTS">
    <checkitem desc="Client Change Option for Production Systems" id="BL_S-3.01">
      <compliant>MANDT like '%'  and ((CCCATEGORY = 'P'  and CCCORACTIV = '2'  and CCNOCLIIND = '3') OR (CCCATEGORY != 'P')) </compliant>
      <complianttext/>
      <noncompliant>MANDT like '%'  and (CCCATEGORY = 'P'  and not ( CCCORACTIV = '2'  and CCNOCLIIND = '3'  ))</noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
  <configstore name="GLOBAL">
    <checkitem desc="Global Modifiable Change Option for Production Systems" id="BL_S-3.02" system_roles="PROD:DEV:TEST:SANDBOX:QA::">
      <compliant>(NAME = 'GLOBAL_SETTING'  and VALUE = 'NOT MODIFIABLE'  and ptree.value = 'PROD' and ptree.hier_id = '1') OR (ptree.value != 'PROD' and ptree.hier_id = '1')</compliant>
      <complianttext/>
      <noncompliant>NAME = 'GLOBAL_SETTING'  and not ( VALUE = 'NOT MODIFIABLE'  ) and ptree.value = 'PROD' and ptree.hier_id = '1'</noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
</targetsystem>
