<?xml version="1.0" encoding="utf-16"?>
<!--
This FRUN CSA policy contains rules derived from SAP Security Baseline Template Version 3
Available at: http://support.sap.com/sos

Target system BL_S-8 / Message Server Security

Version: 001
Date:    Feb 02 2018
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="Message Server Security" id="BL_S-8" multisql="Yes" xsi:schemaLocation="csa_policy.xsd">
  <configstore name="ABAP_INSTANCE_PAHI" sci_id="">
<!--       
 <checkitem desc="File with access control list for message server" id="BL_S-8.01">
   <compliant>NAME = 'ms/acl_info'  and VALUE like 'your mount path/acl file name' </compliant>
   <complianttext/>
   <noncompliant>NAME = 'ms/acl_info'  and not ( VALUE like 'your mount path/acl file name'  ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
-->
    <checkitem desc="Deactivated administration port for external clients" id="BL_S-8.02">
      <compliant>NAME = 'ms/admin_port'  and (replace_regexpr ('^$' IN VALUE with 0) &lt;= 0) </compliant>
      <complianttext/>
      <noncompliant>NAME = 'ms/admin_port'  and not ((replace_regexpr ('^$' IN VALUE with 0) &lt;= 0)) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="Prohibite external monitor" id="BL_S-8.03">
      <compliant>NAME = 'ms/monitor'  and VALUE = '0' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'ms/monitor'  and not ( VALUE = '0'  ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="Defined internal port for server communication" id="BL_S-8.04">
      <compliant>NAME = 'rdisp/msserv_internal'  and VALUE not like '' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'rdisp/msserv_internal'  and not ( VALUE not like ''  ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
  <configstore name="MS_SECINFO" name_extended="" sci_id="" system_type="ABAP">
    <checkitem desc="Message server hardening (acl_info content)" id="BL_S-8.05" operator="">
      <compliant>line like '%' and text like_regexpr 'HOST=[^*]'</compliant>
      <noncompliant>line like '%' and text = 'HOST=*'</noncompliant>
    </checkitem>
  </configstore>
</targetsystem>
