<?xml version="1.0" encoding="utf-8"?>
<!-- 
Reference:   SAP Security Baseline Template v2.4 (2022/12)
Link:        https://support.sap.com/content/dam/support/en_us/library/ssp/offerings-and-programs/support-services/sap-security-optimization-services-portfolio/Security_Baseline_Template_V2.zip
Version:     001
Date:        2023-04-28
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL2.4] Secure Network Configuration (BTP)" id="2PNETCF" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <configstore name="SCC_CONFIG">
    <checkitem desc="[p2-STANDARD] SCC Parameter: isHaActive (SAP Cloud Connector)" id="NETCF-P_a">
      <compliant>
        NAME = 'isHaActive' AND VALUE='true'
      </compliant>
      <noncompliant>
        NAME = 'isHaActive' AND NOT VALUE='true'
      </noncompliant>
    </checkitem>
  </configstore>
</targetsystem>