<?xml version="1.0" encoding="utf-16"?>
<!--
This FRUN CSA policy contains rules derived from SAP Security Baseline Template Version 3
Available at: http://support.sap.com/sos

Target system BL_O-2 / No use of authorization profiles SAP_ALL 

Version: 001
Date:    Feb 02 2018
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="Users with critical Profiles (ABAP)" id="BL_O-2" multisql="Yes" xsi:schemaLocation="csa_policy.xsd">
  <configstore name="AUTH_PROFILE_USER">
    <checkitem desc="USERNAMES with SAP_ALL profile" id="BL_O-2.01">
      <compliant>PROFILE = 'SAP_ALL'  and USERNAME in ( 'DDIC', 'SAP*')  and STATUS = 'USER_IS_AUTHORIZED' </compliant>
      <complianttext/>
      <noncompliant>PROFILE = 'SAP_ALL'  and not ( USERNAME in ( 'DDIC', 'SAP*') )  and STATUS = 'USER_IS_AUTHORIZED'</noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
</targetsystem>
