<?xml version="1.0" encoding="utf-8"?>
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL 2.2] Message Server Security - ABAP" id="BL22_MSGSRV-A" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <configstore name="ABAP_INSTANCE_PAHI" sci_id="">
    <checkitem desc="[p2-STANDARD] Defined internal port for server communication" id="MSGSRV-A_a">
      <compliant>NAME = 'rdisp/msserv_internal'  and lpad(VALUE, 4, '0') &gt; '0000' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'rdisp/msserv_internal'  and not ( lpad(VALUE, 4, '0') &gt; '0000' ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <!-- Alternative Syntax:
    <checkitem desc="[p2-STANDARD] Defined internal port for server communication" id="MSGSRV-A_a">
      <compliant>NAME = 'rdisp/msserv_internal'  and VALUE not like_regexpr '^\s*$' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'rdisp/msserv_internal'  and VALUE like_regexpr '^\s*$' </noncompliant>
      <noncomplianttext/>
    </checkitem>
    -->
    <checkitem desc="[p2-STANDARD] Prohibite external monitor" id="MSGSRV-A_b">
      <compliant>NAME = 'ms/monitor'  and VALUE = '0' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'ms/monitor'  and not ( VALUE = '0'  ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Deactivated administration port for external clients" id="MSGSRV-A_c">
      <compliant>NAME = 'ms/admin_port'  and lpad(VALUE, 4, '0') &lt;= '0000' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'ms/admin_port'  and not (lpad(VALUE, 4, '0') &lt;= '0000') </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p1-CRITICAL] File with access control list for message server" id="MSGSRV-A_d">
      <compliant>    NAME = 'ms/acl_info' and 
                     ( VALUE like_regexpr '^/usr/sap/(\w{3})/SYS/global/ms_acl_info' or
                       VALUE like_regexpr '^/sapmnt/(\w{3})/SYS/global/ms_acl_info'  or
                       VALUE like_regexpr '^\w:\\usr\\sap\\(\w{3})\\SYS\\global\\ms_acl_info(\.DAT|\.dat)?' )
      </compliant>
      <complianttext/>
      <noncompliant> NAME = 'ms/acl_info' and not 
                     ( VALUE like_regexpr '^/usr/sap/(\w{3})/SYS/global/ms_acl_info' or
                       VALUE like_regexpr '^/sapmnt/(\w{3})/SYS/global/ms_acl_info'  or
                       VALUE like_regexpr '^\w:\\usr\\sap\\(\w{3})\\SYS\\global\\ms_acl_info(\.DAT|\.dat)?' ) 
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <!-- Alternative rule (more relaxed):
    <checkitem desc="[p1-CRITICAL] File with access control list for message server" id="BL_MSGSRV-A_d">
      <compliant>    NAME = 'ms/acl_info' and VALUE like_regexpr '.*\w{3}.SYS.global.ms_acl_info.*' </compliant>
      <complianttext/>
      <noncompliant> NAME = 'ms/acl_info' and not VALUE like_regexpr '.*\w{3}.SYS.global.ms_acl_info.*'  </noncompliant>
      <noncomplianttext/>
    </checkitem>
    -->
  </configstore>
  <!--	MSGSRV-A: Message Server Security - ABAP-->
  <configstore name="MS_SECINFO" name_extended="" sci_id="" system_type="ABAP">
    <checkitem desc="[p1-CRITICAL] Message server hardening (acl_info content)" id="MSGSRV-A_e.1" operator="">
      <compliant>    line like '%' and text like_regexpr '.*HOST=[^*].*' </compliant>
      <noncompliant> line like '%' and not text like_regexpr '.*HOST=[^*].*'  </noncompliant>
    </checkitem>
<!--
    <checkitem desc="[p1-CRITICAL] Message server hardening (acl_info content)" id="MSGSRV-A_e.2" operator="">
      <compliant>    line like '%' and not text like_regexpr 'HOST=[*]' </compliant>
      <noncompliant> line like '%' and text like_regexpr 'HOST=[*]'  </noncompliant>
    </checkitem>
-->
  </configstore>
</targetsystem>