<?xml version="1.0" encoding="utf-8"?>
<!-- This policy provides check items for the SAP Security Baseline Template 2.1, 
Chapter:     PROCESS - ABAP

Comments:
  Please review and adapt the check according to your needs.
  Consider the following:
  1.  Checks for the Audit slots of users SAP*, DDIC and SAPCPIC are split into 2 checks:
       1.1 Check if an audit slot for this user exists at all.
       1.2 Check if the audit slot for this user is fully compliant with SAP recommendation 
  2. Checks for additional critical users need to be adjusted based on your naming convention.
       For cases where no naming convention for users exists, check id AUDIT-A_b2.1 and AUDIT-A_b2.2 check if an appropriate audit policy for additional critical users exists at all
       For cases where a naming convention like SAPSUPPORT* or FF* is used, zhe check id AUDIT-A_b2.1a and AUDIT-A_b2.2b can be used or adjusted.
  3. Checks on specific audit events are usually not recommended because patterns of a message vector may not stable when more than required events are selected.
      Specify message vectors only if exact configuration shall be enforced.
  4. Checks on the auditing of system events do not need to be monitored because the auditing of these events are part of SAP's Secure Default and cannot be switched off in current product versions.
  5. Checks on current values are only partially available (2 checks) due to missing data in ST-A/PI Data Collector (9.10.2020). 
      Check id get a "p" appended when referring to Instance Profile Parameters (AUDIT-A_a.1p). Note that for this case Kernel Parameters may overwrite checked values.

Version:    001
Date:       Oct 2020
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL 2.2] PROCESS - ABAP" id="BL22_PROCESS-A" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <!-- 2.4.2	DATA PROTECTION AND PRIVACY -->
  <configstore name="ABAP_SECSTORE_INFO">
    <checkitem desc="[p3-EXTENDED] SECSTORE individual master key)" id="SECSTO-A">
      <compliant>    NAME like 'Encryption%' and  VALUE like '%OK%' </compliant>
      <noncompliant> NAME like 'Encryption%' and not VALUE like '%OK%' </noncompliant>
    </checkitem>
  </configstore>
  <!-- Section 2.4.3      AUDIT AND FRAUD MANAGEMENT -->
  <!-- KERNEL PARAMETERS for Auditing -->
  <configstore name="AUDIT_CONFIGURATION">
    <checkitem desc="[p2-STANDARD] Audit Log activated (current values)" id="AUDIT-A_a.1">
      <compliant>    NAME = 'Enabled' and VALUE = 'Yes'</compliant>
      <noncompliant> NAME = 'Enabled' and not VALUE = 'Yes'</noncompliant>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Log number of selection slots (current values)" id="AUDIT-A_a.4">
      <compliant>    NAME = 'SlotCount' and lpad(VALUE, 4, '0') &gt;= '0010' </compliant>
      <complianttext/>
      <noncompliant> NAME = 'SlotCount' and not (lpad(VALUE, 4, '0') &gt;= '0010') </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
  <!-- INSTANCE PROFILE PARAMETERS / KERNEL PARAMETERS for Auditing -->
  <!-- Note: The profile parameters are overwritten if kernel parameters are configured.Policy checks the actual runtime value of parameters -->
  <configstore name="ABAP_INSTANCE_PAHI">
    <checkitem desc="[p2-STANDARD] Audit Log activated (Profile Parameter)" id="AUDIT-A_a.1p">
      <compliant>    NAME = 'rsau/enable' and VALUE = '1'</compliant>
      <noncompliant> NAME = 'rsau/enable' and VALUE != '1' </noncompliant>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Log integrity protection (Profile Parameter)" id="AUDIT-A_a.2p">
      <compliant>    NAME = 'rsau/integrity' and VALUE = '1' </compliant>
      <noncompliant> NAME = 'rsau/integrity' and VALUE != '1' </noncompliant>
    </checkitem>
    <checkitem desc="[p2-STANDARD]  Audit Log IP address logging (Profile Parameter)" id="AUDIT-A_a.3p">
      <compliant>    NAME = 'rsau/log_peer_address' and VALUE = '1' </compliant>
      <noncompliant> NAME = 'rsau/log_peer_address' and NOT (VALUE = '1') </noncompliant>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Log number of selection slots (Profile Parameter)" id="AUDIT-A_a.4p">
      <compliant>    NAME = 'rsau/selection_slots' and lpad(VALUE, 4, '0') &gt;= '0010' </compliant>
      <complianttext/>
      <noncompliant> NAME = 'rsau/selection_slots' and not (lpad(VALUE, 4, '0') &gt;= '0010') </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Log user selection method used inside SAP kernel (Profile Parameter)" id="AUDIT-A_a.5p">
      <compliant>    NAME = 'rsau/user_selection' and VALUE = '1' </compliant>
      <noncompliant> NAME = 'rsau/user_selection' and VALUE != '1' </noncompliant>
    </checkitem>
  </configstore>
  <!-- AUDIT SLOTS -->
  <configstore name="AUDIT_CONFIGURATION_SLOT">
    <checkitem desc="[p2-STANDARD] Audit Slot for SAP* user exists (Filter SAP#*)" id="AUDIT-A_b1.1" operator="EXIST">
      <compliant>
                       SLOTINDEX like '%'
                       and STATUS = 'X'
                       and UNAME = 'SAP#*' 
      </compliant>
      <noncompliant>
                       SLOTINDEX like '%'  
                       and STATUS = 'X'
                       and UNAME = 'SAP#*' 
      </noncompliant>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot for SAP* user specified as recommended" id="AUDIT-A_b1.2">
      <compliant>
                       SLOTINDEX like '%'
                       and UNAME = 'SAP#*' 

                       and STATUS = 'X'
                       and CLIENT = '*' 
                       
                       and LOWBUTTON = 'X'

                       and LOGIN = 'X'
                       and RFCLOGIN = 'X'
                       and TASTART = 'X'
                       and REPOSTART = 'X'
                       and USERMASTER = 'X'
                       and RFCSTART = 'X'
                       and MISC = 'X'                          
      </compliant>
      <noncompliant>
                       SLOTINDEX like '%'  
                       and UNAME = 'SAP#*' 
                       
                       and (
                         STATUS != 'X'
                         or CLIENT != '*'

                         or LOWBUTTON != 'X'

                         or RFCLOGIN != 'X'
                         or TASTART != 'X'
                         or REPOSTART != 'X'
                         or USERMASTER != 'X'
                         or RFCSTART != 'X'
                         or MISC != 'X'
                       )
      </noncompliant>
      <joinstore name="AUDIT_CONFIGURATION_SLOT" no_data_found="Yes" not_in_output="Yes">
        <joincompliant>    SLOTINDEX like '%'  and UNAME = 'SAP#*' 
        </joincompliant>
        <joinnoncompliant> SLOTINDEX like '%'  and UNAME = 'SAP#*'  
        </joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot for DDIC user exists" id="AUDIT-A_b1.3" operator="EXIST">
      <compliant>
                       SLOTINDEX like '%'
                       and STATUS = 'X'
                       and UNAME = 'DDIC' 
      </compliant>
      <noncompliant>
                       SLOTINDEX like '%'  
                       and STATUS = 'X'
                       and UNAME = 'DDIC' 
      </noncompliant>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot for DDIC user specified as recommended" id="AUDIT-A_b1.4">
      <compliant>
                       SLOTINDEX like '%'
                       and UNAME = 'DDIC' 

                       and STATUS = 'X'
                       and CLIENT = '*' 
                       
                       and ( MEDBUTTON = 'X' or LOWBUTTON = 'X' )

                       and LOGIN = 'X'
                       and RFCLOGIN = 'X'
                       and TASTART = 'X'
                       and REPOSTART = 'X'
                       and USERMASTER = 'X'
                       and RFCSTART = 'X'
                       and MISC = 'X'                          
      </compliant>
      <noncompliant>
                       SLOTINDEX like '%'  
                       and UNAME = 'DDIC' 
                       
                       and (
                         STATUS != 'X'
                         or CLIENT != '*'

                         or ( MEDBUTTON != 'X' and LOWBUTTON != 'X' )

                         or RFCLOGIN != 'X'
                         or TASTART != 'X'
                         or REPOSTART != 'X'
                         or USERMASTER != 'X'
                         or RFCSTART != 'X'
                         or MISC != 'X'
                       )
      </noncompliant>
      <joinstore name="AUDIT_CONFIGURATION_SLOT" no_data_found="Yes" not_in_output="Yes">
        <joincompliant>    SLOTINDEX like '%' and STATUS = 'X' and UNAME = 'DDIC' </joincompliant>
        <joinnoncompliant> SLOTINDEX like '%' and STATUS = 'X' and UNAME = 'DDIC'  </joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot  for SAPCPIC user exists" id="AUDIT-A_b1.5" operator="EXIST">
      <compliant>    SLOTINDEX like '%' and STATUS = 'X' and UNAME = 'SAPCPIC'  </compliant>
      <noncompliant> SLOTINDEX like '%' and STATUS = 'X' and UNAME = 'SAPCPIC'  </noncompliant>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot for SAPCPIC user specified as recommended" id="AUDIT-A_b1.6">
      <compliant>
                       SLOTINDEX like '%'
                       and UNAME = 'SAPCPIC' 

                       and STATUS = 'X'
                       and CLIENT = '*' 
                       
                       and ( MEDBUTTON = 'X' or LOWBUTTON = 'X' )

                       and LOGIN = 'X'
                       and RFCLOGIN = 'X'
                       and TASTART = 'X'
                       and REPOSTART = 'X'
                       and USERMASTER = 'X'
                       and RFCSTART = 'X'
                       and MISC = 'X'                          
      </compliant>
      <noncompliant>
                       SLOTINDEX like '%'  
                       and UNAME = 'SAPCPIC' 
                       
                       and (
                         STATUS != 'X'
                         or CLIENT != '*'

                         or ( MEDBUTTON != 'X' and LOWBUTTON != 'X' )

                         or RFCLOGIN != 'X'
                         or TASTART != 'X'
                         or REPOSTART != 'X'
                         or USERMASTER != 'X'
                         or RFCSTART != 'X'
                         or MISC != 'X'
                       )
      </noncompliant>
      <joinstore name="AUDIT_CONFIGURATION_SLOT" no_data_found="Yes" not_in_output="Yes">
        <joincompliant>    SLOTINDEX like '%' and STATUS = 'X' and UNAME = 'SAPCPIC' </joincompliant>
        <joinnoncompliant> SLOTINDEX like '%' and STATUS = 'X' and UNAME = 'SAPCPIC'  </joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot for CRITICAL USERS exists" id="AUDIT-A_b2.1" operator="EXIST">
      <compliant>
                       SLOTINDEX like '%' 
                       and STATUS = 'X'
                       and ( UNAME != 'SAP#*' and UNAME != 'DDIC' and UNAME != 'SAPCPIC' and UNAME != '*' )  
      </compliant>
      <noncompliant>
                       SLOTINDEX like '%' 
                       and STATUS = 'X'
                       and ( UNAME != 'SAP#*' and UNAME != 'DDIC' and UNAME != 'SAPCPIC'  and UNAME != '*' ) 
      </noncompliant>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot for CRITICAL USERS correctly specified" id="AUDIT-A_b2.2">
      <compliant>
                       SLOTINDEX like '%'
                       and ( UNAME != 'SAP#*' and UNAME != 'DDIC' and UNAME != 'SAPCPIC' and UNAME != '*' ) 

                       and STATUS = 'X'
                       and CLIENT = '*' 
                       
                       and ( LOWBUTTON = 'X' )

                       and LOGIN = 'X'
                       and RFCLOGIN = 'X'
                       and TASTART = 'X'
                       and REPOSTART = 'X'
                       and USERMASTER = 'X'
                       and RFCSTART = 'X'
                       and MISC = 'X'                          
      </compliant>
      <noncompliant>
                       SLOTINDEX like '%'  
                       and ( UNAME != 'SAP#*' and UNAME != 'DDIC' and UNAME != 'SAPCPIC' and UNAME != '*' ) 
                       
                       and (
                         STATUS != 'X'
                         or CLIENT != '*'

                         or LOWBUTTON != 'X'

                         or RFCLOGIN != 'X'
                         or TASTART != 'X'
                         or REPOSTART != 'X'
                         or USERMASTER != 'X'
                         or RFCSTART != 'X'
                         or MISC != 'X'
                       )
      </noncompliant>
      <joinstore name="AUDIT_CONFIGURATION_SLOT" no_data_found="Yes" not_in_output="Yes">
        <joincompliant>    SLOTINDEX like '%'
                           and STATUS = 'X'  
                           and ( UNAME != 'SAP#*' and UNAME != 'DDIC' and UNAME != 'SAPCPIC' and UNAME != '*' ) 
        </joincompliant>
        <joinnoncompliant> SLOTINDEX like '%'  
                           and STATUS = 'X'
                           and ( UNAME != 'SAP#*' and UNAME != 'DDIC' and UNAME != 'SAPCPIC' and UNAME != '*' ) 
        </joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot for SUPPORT or FIREFIGHTER Users exists" id="AUDIT-A_b2.1a" operator="EXIST">
      <compliant>
                       SLOTINDEX like '%' 
                       and STATUS = 'X'
                       and ( UNAME like 'SAPSUPPORT%' or UNAME like 'FF%' )  
      </compliant>
      <noncompliant>
                       SLOTINDEX like '%' 
                       and STATUS = 'X'
                       and ( UNAME like 'SAPSUPPORT%' or UNAME like 'FF%' )  
      </noncompliant>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot for SUPPORT or FIREFIGHTER users correctly specified" id="AUDIT-A_b2.2a">
      <compliant>
                       SLOTINDEX like '%'
                       and ( UNAME like 'SAPSUPPORT%' or UNAME like 'FF%' )  

                       and STATUS = 'X'
                       and CLIENT = '*' 
                       
                       and ( LOWBUTTON = 'X' )

                       and LOGIN = 'X'
                       and RFCLOGIN = 'X'
                       and TASTART = 'X'
                       and REPOSTART = 'X'
                       and USERMASTER = 'X'
                       and RFCSTART = 'X'
                       and MISC = 'X'                          
      </compliant>
      <noncompliant>
                       SLOTINDEX like '%'  
                       and ( UNAME like 'SAPSUPPORT%' or UNAME like 'FF%' )  
                       
                       and (
                         STATUS != 'X'
                         or CLIENT != '*'

                         or LOWBUTTON != 'X'

                         or RFCLOGIN != 'X'
                         or TASTART != 'X'
                         or REPOSTART != 'X'
                         or USERMASTER != 'X'
                         or RFCSTART != 'X'
                         or MISC != 'X'
                       )
      </noncompliant>
      <joinstore name="AUDIT_CONFIGURATION_SLOT" no_data_found="Yes" not_in_output="Yes">
        <joincompliant>    SLOTINDEX like '%'
                           and STATUS = 'X'  
                           and ( UNAME like 'SAPSUPPORT%' or UNAME like 'FF%' )  
        </joincompliant>
        <joinnoncompliant> SLOTINDEX like '%'  
                           and STATUS = 'X'
                           and ( UNAME like 'SAPSUPPORT%' or UNAME like 'FF%' )  
        </joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p2-STANDARD] Audit Slot for ANY USER exists" id="AUDIT-A_b3.1" operator="EXIST">
      <compliant>
                    SLOTINDEX like '%'
                    and UNAME = '*' 
                    and STATUS = 'X'
                    and CLIENT = '*'
                    
                    and LOWBUTTON like '%'
                    and MEDBUTTON like '%'
                    and HIGHBUTTON like '%'
                    
                    and LOGIN like '%'
                    and RFCLOGIN like '%'
                    and TASTART like '%'
                    and REPOSTART like '%'
                    and USERMASTER like '%'
                    and RFCSTART like '%'
                    and MISC like '%' 
                    
                    and SELVAR like '%'
                    and MSGVECT like '%'
      </compliant>
      <noncompliant>
                    SLOTINDEX like '%'
                    and UNAME = '*' 
                    and STATUS = 'X'
                    and CLIENT = '*' 
                    
                    and LOWBUTTON like '%'
                    and MEDBUTTON like '%'
                    and HIGHBUTTON like '%'
                    
                    and LOGIN like '%'
                    and RFCLOGIN like '%'
                    and TASTART like '%'
                    and REPOSTART like '%'
                    and USERMASTER like '%'
                    and RFCSTART like '%'
                    and MISC like '%'
                    
                    and SELVAR like '%'
                    and MSGVECT like '%'
      </noncompliant>
    </checkitem>
  </configstore>
</targetsystem>