<?xml version="1.0" encoding="utf-8"?>
<!-- 
Reference:   SAP Security Baseline Template v2.4 (2022/12)
Link:        https://support.sap.com/content/dam/support/en_us/library/ssp/offerings-and-programs/support-services/sap-security-optimization-services-portfolio/Security_Baseline_Template_V2.zip
Version:     001
Date:        2023-04-28
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL2.4] Session Protection (JAVA)" id="2JSESS" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <!-- SESS-J: Session Protection â€“ Java -->
  <configstore name="http">

    <checkitem desc="[p2-STANDARD] SystemCookiesDataProtection" id="SESS-J_a">
      <compliant>    NAME = 'SystemCookiesDataProtection'  and VALUE = 'true' </compliant>
      <complianttext/>
      <noncompliant> NAME = 'SystemCookiesDataProtection'  and VALUE = 'true'  </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p2-STANDARD] SystemCookiesDataProtection" id="SESS-J_b">
      <compliant>    NAME = 'SystemCookiesHTTPSProtection'  and VALUE = 'true' </compliant>
      <complianttext/>
      <noncompliant> NAME = 'SystemCookiesHTTPSProtection'  and VALUE = 'true'  </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
</targetsystem>