<?xml version="1.0" encoding="utf-16"?>
<!--
This FRUN CSA policy contains rules derived from SAP Security Baseline Template Version 3
Available at: http://support.sap.com/sos

Target system BL_S-7 / RFC Gateway Security

Version: 001
Date:    Feb 02 2018
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="RFC Gateway Security" id="BL_S-7" multisql="Yes" xsi:schemaLocation="csa_policy.xsd">
  <configstore name="ABAP_INSTANCE_PAHI">
    <checkitem desc="Mode for non existing ACL file" id="BL_S-7.01">
      <compliant>NAME = 'gw/acl_mode'  and VALUE = '1' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'gw/acl_mode'  and not ( VALUE = '1'  ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="Enable gateway monitor commands" id="BL_S-7.02">
      <compliant>NAME = 'gw/monitor'  and VALUE = '1' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'gw/monitor'  and not ( VALUE = '1'  ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="Gateway security options" id="BL_S-7.03">
      <compliant>NAME = 'gw/reg_no_conn_info'  and ( VALUE = '1'  or VALUE = '3'  or VALUE = '7'  or VALUE = '15'  or VALUE = '31'  or VALUE = '63'  or VALUE = '127'  or VALUE = '255' )</compliant>
      <complianttext/>
      <noncompliant>NAME = 'gw/reg_no_conn_info'  and not ( ( VALUE = '1'  or VALUE = '3'  or VALUE = '7'  or VALUE = '15'  or VALUE = '31'  or VALUE = '63'  or VALUE = '127'  or VALUE = '255' ) ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="Disable simulation mode for reg_info and sec_info" id="BL_S-7.04">
      <compliant>NAME = 'gw/sim_mode'  and VALUE = '0' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'gw/sim_mode'  and not ( VALUE = '0'  ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
  <configstore name="Parameters">
    <checkitem desc="Mode for non existing ACL file" id="BL_S-7.05">
      <compliant>NAME = 'gw/acl_mode'  and VALUE = '1' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'gw/acl_mode'  and not ( VALUE = '1'  ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="Enable gateway monitor commands" id="BL_S-7.06">
      <compliant>NAME = 'gw/monitor'  and VALUE = '1' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'gw/monitor'  and not ( VALUE = '1'  ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="Gateway security options" id="BL_S-7.07">
      <compliant>NAME = 'gw/reg_no_conn_info'  and ( VALUE = '1'  or VALUE = '3'  or VALUE = '7'  or VALUE = '15'  or VALUE = '31'  or VALUE = '63'  or VALUE = '127'  or VALUE = '255' )</compliant>
      <complianttext/>
      <noncompliant>NAME = 'gw/reg_no_conn_info'  and not ( ( VALUE = '1'  or VALUE = '3'  or VALUE = '7'  or VALUE = '15'  or VALUE = '31'  or VALUE = '63'  or VALUE = '127'  or VALUE = '255' ) ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="Disable simulation mode for reg_info and sec_info" id="BL_S-7.08">
      <compliant>NAME = 'gw/sim_mode'  and VALUE = '0' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'gw/sim_mode'  and not ( VALUE = '0'  ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
  <configstore name="GW_SECINFO" name_extended="" sci_id="" system_type="ABAP">
    <checkitem desc="RFC gateway hardening (secinfo content)" id="BL_S-7.09" operator="">
      <compliant>line like '%' and ( (text like_regexpr 'USER-HOST=[^*]' and text like_regexpr '[^-]HOST=[^*]') or text like_regexpr 'HOST=[*][.]')</compliant>
      <noncompliant>line like '%' and not ( (text like_regexpr 'USER-HOST=[^*]' and text like_regexpr '[^-]HOST=[^*]') or text like_regexpr 'HOST=[*][.]')</noncompliant>
    </checkitem>
  </configstore>
  <configstore name="GW_REGINFO" name_extended="" sci_id="" system_type="ABAP">
    <checkitem desc="RFC gateway hardening (reginfo content)" id="BL_S-7.10" operator="">
      <compliant>line like '%' and ( TEXT like_regexpr 'HOST=[^*]' )</compliant>
      <noncompliant> (line like '%' and TEXT like  '%HOST=[*]%' ) OR ( line like '%' and TEXT = 'P TP=*' )</noncompliant>
    </checkitem>
  </configstore>
  <!--
  <configstore name="GW_PRXINFO" name_extended="" sci_id="" system_type="ABAP">
    <checkitem desc="RFC gateway hardening (prxyinfo content)" id="BL_S-7.11" operator="">
      <compliant>line like '%' and ( text like  '%P SOURCE=local,internal%' or text like '%SOURCE=*%'  )</compliant>
      <noncompliant>line like '%' and ( not ( text like  '%P SOURCE=local,internal%' or text like '%SOURCE=*%' ) or text = 'gw/prxy_info file not found')</noncompliant>
    </checkitem>
  </configstore>
  -->
</targetsystem>
