<?xml version="1.0" encoding="utf-16"?>
<!--
This FRUN CSA policy contains rules derived from SAP Security Baseline Template Version 3
Available at: http://support.sap.com/sos

Target system BL_S-5 / JAVA UME Parameters

Version: 001
Date:    Feb 02 2018
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="UME Parameters" id="BL_S-5" multisql="Yes" xsi:schemaLocation="csa_policy.xsd">
 <configstore  name="com.sap.security.core.ume.service"  >
  <checkitem desc="SAP logon ticket lifetime" id="BL_S-5.01"   >
   <compliant>NAME = 'login.ticket_lifetime'  and (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &lt;= 8 </compliant>
   <complianttext/>
   <noncompliant>NAME = 'login.ticket_lifetime'  and not ( (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &lt;= 8  ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
  <checkitem desc="SAP logon ticket set to HttpOnly" id="BL_S-5.02"   >
   <compliant>NAME = 'ume.logon.httponlycookie'  and ( VALUE = 'true'  or VALUE = 'TRUE' )</compliant>
   <complianttext/>
   <noncompliant>NAME = 'ume.logon.httponlycookie'  and not ( ( VALUE = 'true'  or VALUE = 'TRUE' ) )</noncompliant>
   <noncomplianttext/>
  </checkitem>
  <checkitem desc="Send SAP logon ticket only via HTTPS" id="BL_S-5.03"   >
   <compliant>NAME = 'ume.logon.security.enforce_secure_cookie'  and ( VALUE = 'true'  or VALUE = 'TRUE' )</compliant>
   <complianttext/>
   <noncompliant>NAME = 'ume.logon.security.enforce_secure_cookie'  and not ( ( VALUE = 'true'  or VALUE = 'TRUE' ) ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
  <checkitem desc="Disable old password in password allowed" id="BL_S-5.04"   >
   <compliant>NAME = 'ume.logon.security_policy.oldpass_in_newpass_allowed'  and ( VALUE = 'false'  or VALUE = 'FALSE' )</compliant>
   <complianttext/>
   <noncompliant>NAME = 'ume.logon.security_policy.oldpass_in_newpass_allowed'  and not ( ( VALUE = 'false'  or VALUE = 'FALSE' ) ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
  <checkitem desc="Minimum number of letters and numbers in password" id="BL_S-5.05"   >
   <compliant>NAME = 'ume.logon.security_policy.password_alpha_numeric_required'  and (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &gt;= 1 </compliant>
   <complianttext/>
   <noncompliant>NAME = 'ume.logon.security_policy.password_alpha_numeric_required'  and not ( (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &gt;= 1  ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
  <checkitem desc="Minimum Password Length" id="BL_S-5.06"   >
   <compliant>NAME = 'ume.logon.security_policy.password_min_length'  and (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &gt;= 8 </compliant>
   <complianttext/>
   <noncompliant>NAME = 'ume.logon.security_policy.password_min_length'  and not ( (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &gt;= 8  ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
  <checkitem desc="Minimum number of uppercase and lowercase letters in password required" id="BL_S-5.07"   >
   <compliant>NAME = 'ume.logon.security_policy.password_mix_case_required'  and (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &gt;= 1 </compliant>
   <complianttext/>
   <noncompliant>NAME = 'ume.logon.security_policy.password_mix_case_required'  and not ( (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &gt;= 1  ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
  <checkitem desc="Minimum number of special characters in passwords" id="BL_S-5.08"   >
   <compliant>NAME = 'ume.logon.security_policy.password_special_char_required'  and (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &gt;= 0 </compliant>
   <complianttext/>
   <noncompliant>NAME = 'ume.logon.security_policy.password_special_char_required'  and not ( (SUBSTR_REGEXPR('([[:digit:]]{1,99})' IN VALUE)) &gt;= 0  ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
  <checkitem desc="Disable user ID as part of password" id="BL_S-5.09"   >
   <compliant>NAME = 'ume.logon.security_policy.userid_in_password_allowed'  and ( VALUE = 'false'  or VALUE = 'FALSE' )</compliant>
   <complianttext/>
   <noncompliant>NAME = 'ume.logon.security_policy.userid_in_password_allowed'  and not ( ( VALUE = 'false'  or VALUE = 'FALSE' ) ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
  <checkitem desc="Disable UME Self Registration" id="BL_S-5.10"   >
   <compliant>NAME = 'ume.logon.selfreg'  and ( VALUE = 'false'  or VALUE = 'FALSE' )</compliant>
   <complianttext/>
   <noncompliant>NAME = 'ume.logon.selfreg'  and not ( ( VALUE = 'false'  or VALUE = 'FALSE' ) ) </noncompliant>
   <noncomplianttext/>
  </checkitem>
 </configstore>
</targetsystem>

