<?xml version="1.0" encoding="utf-8"?>
<!--
Reference:   SAP Security Baseline Template v2.4 (2022/12)
Link:        https://support.sap.com/content/dam/support/en_us/library/ssp/offerings-and-programs/support-services/sap-security-optimization-services-portfolio/Security_Baseline_Template_V2.zip
Version:     002
Date:        2024-02-20
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL2.4] Password Policy (HANA)" id="1HPWDPOL" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <configstore name="HDB_PARAMETER">
    <!-- PWDPOL-H: Password Policy - HANA (SystemDB) -->
    <checkitem desc="[p1-CRITICAL] Minimal password length" id="PWDPOL-H_a.1" system_attributes="DB_SYSTEMDB:TRUE">
      <compliant>    FILE_NAME = 'nameserver.ini' and HOST like '%'  and 
                     SECT = 'password policy'  and 
                     NAME = 'minimal_password_length'  and 
                     LAYER like '%'  and 
                     lpad(VALUE,4,'0') &gt;= '0008' 
      </compliant>
      <complianttext/>
      <noncompliant> FILE_NAME = 'nameserver.ini' and HOST like '%'  and 
                     SECT = 'password policy'  and 
                     NAME = 'minimal_password_length'  and 
                     LAYER like '%'  and 
                     NOT lpad(VALUE,4,'0') &gt;= '0008'
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p1-CRITICAL] Unused initial Password Lifetime" id="PWDPOL-H_a.2" system_attributes="DB_SYSTEMDB:TRUE">
      <compliant>    FILE_NAME = 'nameserver.ini' and HOST like '%'  and 
                     SECT = 'password policy'  and (NAME = 'maximum_unused_inital_password_lifetime' OR NAME = 'maximum_unused_initial_password_lifetime') and 
                     LAYER like '%'  and 
                     lpad(VALUE,4,'0') between '0001' and '0014'  
      </compliant>
      <complianttext/>
      <noncompliant> FILE_NAME = 'nameserver.ini' and HOST like '%'  and 
                     SECT = 'password policy'  and (NAME = 'maximum_unused_inital_password_lifetime' OR NAME = 'maximum_unused_initial_password_lifetime') and 
                     LAYER like '%'  and  
                     NOT lpad(VALUE,4,'0') between '0001' and '0014'  
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p1-CRITICAL] Unused Productive Password Lifetime" id="PWDPOL-H_c" system_attributes="DB_SYSTEMDB:TRUE">
      <compliant>    FILE_NAME = 'nameserver.ini' and 
                     SECT = 'password policy' and 
                     NAME = 'maximum_unused_productive_password_lifetime' and  
                     lpad(VALUE,4,'0') between '0001' and '0090' and 
                     LAYER like '%' and HOST like '%'
      </compliant>
      <noncompliant> FILE_NAME = 'nameserver.ini' and 
                     SECT = 'password policy' and 
                     NAME = 'maximum_unused_productive_password_lifetime' and not
                     lpad(VALUE,4,'0') between '0001' and '0090' and 
                     LAYER like '%' and HOST like '%'
      </noncompliant>
    </checkitem>
    <!--  PWDPOL-H: Password Policy - HANA (TenantDB) -->
    <checkitem desc="[p1-CRITICAL] Minimal password length" id="PWDPOL-H_a.1" system_attributes="DB_TENANT:TRUE">
      <compliant>    FILE_NAME = 'indexserver.ini' and HOST like '%'  and 
                     SECT = 'password policy'  and 
                     NAME = 'minimal_password_length'  and 
                     LAYER like '%'  and 
                     lpad(VALUE,4,'0') &gt;= '0008' 
      </compliant>
      <complianttext/>
      <noncompliant> FILE_NAME = 'indexserver.ini' and HOST like '%'  and 
                     SECT = 'password policy'  and 
                     NAME = 'minimal_password_length'  and 
                     LAYER like '%'  and 
                     NOT lpad(VALUE,4,'0') &gt;= '0008' 
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p1-CRITICAL] Unused initial Password Lifetime" id="PWDPOL-H_a.2" system_attributes="DB_TENANT:TRUE">
      <compliant>    FILE_NAME = 'indexserver.ini' and HOST like '%' and
                     SECT = 'password policy'  and 
                     (NAME = 'maximum_unused_inital_password_lifetime' OR NAME = 'maximum_unused_initial_password_lifetime')  and 
                     LAYER like '%'  and  
                     lpad(VALUE,4,'0') between '0001' and '0014'  
      </compliant>
      <complianttext/>
      <noncompliant> FILE_NAME = 'indexserver.ini' and HOST like '%' and 
                     SECT = 'password policy'  and 
                     (NAME = 'maximum_unused_inital_password_lifetime' OR NAME = 'maximum_unused_initial_password_lifetime') and 
                     LAYER like '%'  and NOT 
                      lpad(VALUE,4,'0') between '0001' and '0014'  
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p1-CRITICAL] Unused Productive Password Lifetime" id="PWDPOL-H_c" system_attributes="DB_TENANT:TRUE">
      <compliant>    FILE_NAME = 'indexserver.ini' and  HOST like '%' and 
                     SECT = 'password policy' and 
                     NAME = 'maximum_unused_productive_password_lifetime' and 
                     LAYER like '%' and
                     lpad(VALUE,4,'0') between '0001' and '0090' 
      </compliant>
      <noncompliant> FILE_NAME = 'indexserver.ini' and HOST like '%' and
                     SECT = 'password policy' and 
                     NAME = 'maximum_unused_productive_password_lifetime' and 
                     LAYER like '%' and 
                     NOT lpad(VALUE,4,'0') between '0001' and '0090' 
      </noncompliant>
    </checkitem>
  </configstore>
  <configstore name="HDB_USERGROUP_PARAMETERS">
    <checkitem desc="[p1-CRITICAL] Usergroup PWD Policy: First PW Change" id="PWDPOL-H_a1b" not_found="ignore">
      <compliant>    USERGROUP_NAME like '%' and
                     PARAMETER_NAME = 'minimal_password_length' and 
                     PARAMETER_SET_NAME like '%' and
                     IS_PARAMETER_SET_ENABLED like '%' and
                     LPAD(PARAMETER_VALUE,4,'0') &gt;= '0008'
      </compliant>
      <complianttext/>
      <noncompliant> USERGROUP_NAME like '%' and
                     PARAMETER_NAME = 'minimal_password_length' and 
                     PARAMETER_SET_NAME like '%' and
                     IS_PARAMETER_SET_ENABLED like '%' and
                     not LPAD(PARAMETER_VALUE,4,'0') &gt;= '0008'
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p1-CRITICAL] Usergroup PWD Policy: Max Unused initial Password Lifetime" id="PWDPOL-H_a.2b" not_found="ignore">
      <compliant>    USERGROUP_NAME like '%' and
                     PARAMETER_NAME = 'maximum_unused_inital_password_lifetime' and 
                     PARAMETER_SET_NAME like '%' and
                     IS_PARAMETER_SET_ENABLED like '%' and
                     LPAD(PARAMETER_VALUE,4,'0') between '0001' and '0014'
      </compliant>
      <complianttext/>
      <noncompliant> USERGROUP_NAME like '%' and
                     PARAMETER_NAME = 'maximum_unused_inital_password_lifetime' and 
                     PARAMETER_SET_NAME like '%' and
                     IS_PARAMETER_SET_ENABLED like '%' and
                     not LPAD(PARAMETER_VALUE,4,'0') between '0001' and '0014'
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p1-CRITICAL] Usergroup PWD Policy: Max unused Productive Password Lifetime" id="PWDPOL-H_a.3b" not_found="ignore">
      <compliant>    USERGROUP_NAME like '%' and
                     PARAMETER_NAME = 'maximum_unused_productive_password_lifetime' and  
                     PARAMETER_SET_NAME like '%' and
                     IS_PARAMETER_SET_ENABLED like '%' and
                     LPAD(PARAMETER_VALUE,4,'0') between '0001' and '0090'
      </compliant>
      <noncompliant> USERGROUP_NAME like '%' and
                     PARAMETER_NAME = 'maximum_unused_productive_password_lifetime' and 
                     PARAMETER_SET_NAME like '%' and
                     IS_PARAMETER_SET_ENABLED like '%' and
                     not LPAD(PARAMETER_VALUE,4,'0') between '0001' and '0090'
      </noncompliant>
    </checkitem>
  </configstore>
</targetsystem>