<?xml version="1.0" encoding="utf-8"?>
<!--
Reference:   SAP Security Baseline Template v2.4 (2022/12)
Link:        https://support.sap.com/content/dam/support/en_us/library/ssp/offerings-and-programs/support-services/sap-security-optimization-services-portfolio/Security_Baseline_Template_V2.zip
Version:     001
Date:        2023-04-28
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL2.4] Regular Security Updates (HANA)" id="1HSECUPD" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <configstore name="HDB_VERSION">
    <checkitem desc="[p1-CRITICAL] Security Maintenance Status of HANA Version (SPS)" id="SECUPD-H.b">
      <compliant>    (NAME = 'VERSION' AND substring(VALUE,0,7) &gt;= '2.00.05')
      </compliant>
      <noncompliant> ( (NAME = 'VERSION' AND substring(VALUE,0,7) = '2.00.00' )
                    OR (NAME = 'VERSION' AND substring(VALUE,0,7) = '2.00.01')
                    OR (NAME = 'VERSION' AND substring(VALUE,0,7) = '2.00.02')
                    OR (NAME = 'VERSION' AND substring(VALUE,0,7) = '2.00.03')
                    OR (NAME = 'VERSION' AND substring(VALUE,0,7) = '2.00.04')
                    OR (NAME = 'VERSION' AND substring(VALUE,0,4) = '1.00' ) )
      </noncompliant>
    </checkitem>
    <checkitem desc="[p1-CRITICAL] Last detected HANA Update longer than 1 year ago" id="SECUPD-H.a" operator="">
      <compliant>  
        NAME = 'VERSION' and VALUE like '%' and CD_HIST_MOD like '%' 
        and ( CD_HIST_DATE &gt;= to_number(replace(add_days(current_date, -365), '-', '')||replace(current_time, ':', '') ))
    </compliant>
      <noncompliant> 
        NAME = 'VERSION' and VALUE like '%'  and CD_HIST_MOD like '%' 
        and ( CD_HIST_DATE &lt; to_number(replace(add_days(current_date, -365), '-', '')||replace(current_time, ':', '') ))
      </noncompliant>
    </checkitem>
  </configstore>
</targetsystem>