<?xml version="1.0" encoding="utf-8"?>
<!-- 
Reference:   SAP Security Baseline Template v2.4 (2022/12)
Link:        https://support.sap.com/content/dam/support/en_us/library/ssp/offerings-and-programs/support-services/sap-security-optimization-services-portfolio/Security_Baseline_Template_V2.zip
Version:     001
Date:        2023-04-28
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL2.4] Regular Security Updates (ABAP)" id="1ASECUPD" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <configstore name="COMP_LEVEL">
    <checkitem desc="Last detected Update older than 1 year" id="SECUPD-A_a1" not_found="ignore" system_attributes="SYSTEM_TYPE:ABAP">
      <compliant>    COMPONENT like '%' 
                 and VERSION like '%' 
                 and SP like '%'      
                 and CD_HIST_MOD like '%' 
                 and not ( CD_HIST_DATE &lt; to_number(replace(add_days(current_date, -365), '-', '')||replace(current_time, ':', '') ))
      </compliant>
      <noncompliant> COMPONENT like '%' 
                 and VERSION like '%' 
                 and SP like '%'      
                 and CD_HIST_MOD like '%' 
                 and ( CD_HIST_DATE &lt; to_number(replace(add_days(current_date, -365), '-', '')||replace(current_time, ':', '') ))
      </noncompliant>
    </checkitem>
    <checkitem desc="Component Level older then 24 months" id="SECUPD-A_a2" not_found="ignore" system_attributes="SYSTEM_TYPE:ABAP">
      <compliant>    COMPONENT like '%' 
                 and VERSION like '%' 
                 and SP like '%'  
                 and SP_REL_DATE != ''and (add_days(current_date,-730)) &lt; (CASE WHEN SP_REL_DATE like_regexpr '^\d{8,8}$' THEN SP_REL_DATE WHEN SP_REL_DATE = 'NEWER' THEN CURRENT_DATE ELSE '00000000' END)
      </compliant>
      <noncompliant> COMPONENT like '%' 
                 and VERSION like '%'
                 and SP like '%'
                 and SP_REL_DATE != '' and not (add_days(current_date,-730)) &lt; (CASE WHEN SP_REL_DATE like_regexpr '^\d{8,8}$' THEN SP_REL_DATE WHEN SP_REL_DATE = 'NEWER' THEN CURRENT_DATE ELSE '00000000' END)
      </noncompliant>
    </checkitem>
  </configstore>
</targetsystem>