<?xml version="1.0" encoding="utf-8"?>
<!-- 
Reference:   SAP Security Baseline Template v2.4 (2022/12)
Link:        https://support.sap.com/content/dam/support/en_us/library/ssp/offerings-and-programs/support-services/sap-security-optimization-services-portfolio/Security_Baseline_Template_V2.zip
Version:     001
Date:        2023-04-28
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[STANDARD] Directory Traversal Protection (ABAP)" id="2AFILE" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <configstore name="ABAP_INSTANCE_PAHI">
    <!--  FILE-A:  Directory Traversal Protection â€“ ABAP -->
    <!--  You may strengthen the rule about profile parameter abap/path_normalization = ext according to notes 2562089 and 2551541. -->
    <checkitem desc="[p2-STANDARD] abap/path_normalization (Profile Parameter, Directory Traversal Protection)" id="FILE-A_a2">
      <compliant>    NAME = 'abap/path_normalization'  and  VALUE != 'off'   </compliant>
      <complianttext/>
      <noncompliant> NAME = 'abap/path_normalization'  and  VALUE = 'off'    </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
</targetsystem>