<?xml version="1.0" encoding="utf-8"?>
<!-- 
Reference:   SAP Security Baseline Template v2.4 (2022/12)
Link:        https://support.sap.com/content/dam/support/en_us/library/ssp/offerings-and-programs/support-services/sap-security-optimization-services-portfolio/Security_Baseline_Template_V2.zip
Version:     001
Date:        2023-04-28
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL2.4] Change Protection of Prod System (ABAP)" id="3ACHANGE" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <!-- id= CHANGE-A_d, e, f: Transport Parameters available in Focused Run as of FRUN 3.0 FP00 -->
  <configstore name="TRANSPORT_TOOL">
    <checkitem desc="[p3-EXTENDED] Transport Parameter RECCLIENT is activated" id="CHANGE-A_d">
      <compliant>    NAME = 'RECCLIENT'  and (VALUE != 'OFF' and length(VALUE) &gt; 0) </compliant>
      <complianttext/>
      <noncompliant> NAME = 'RECCLIENT'  and (VALUE = 'OFF' or length(VALUE) = 0) </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Transport Parameter VERS_AT_IMP is activated" id="CHANGE-A_e.1" system_roles="DEV">
      <compliant>    NAME = 'VERS_AT_EXP'  and VALUE = '1' </compliant>
      <complianttext/>
      <noncompliant> NAME = 'VERS_AT_EXP'  and not VALUE = '1' </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Transport Parameter VERS_AT_IMP is activated in Prod." id="CHANGE-A_e.2" system_roles="PROD">
      <compliant>    NAME = 'VERS_AT_IMP'  and VALUE = 'ALWAYS'  </compliant>
      <complianttext/>
      <noncompliant> NAME = 'VERS_AT_IMP'  and VALUE != 'ALWAYS' </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Transport Parameter TP_RELEASE is compliant" id="CHANGE-A_g.1">
      <compliant>    NAME = 'TP_RELEASE'  and VALUE &gt;= '380.44.90'  </compliant>
      <complianttext/>
      <noncompliant> NAME = 'TP_RELEASE'  and VALUE &lt; '380.44.90' </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Transport Parameter TP_VERSION is compliant" id="CHANGE-A_g.2">
      <compliant>    NAME = 'TP_VERSION'  and VALUE &gt;= '380'  </compliant>
      <complianttext/>
      <noncompliant> NAME = 'TP_VERSION'  and VALUE &lt; '380' </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Transport Parameter CHK_CRIOBJ_AT_EXPORT is compliant" id="CHANGE-A_h" system_roles="DEV">
      <compliant>    NAME = 'CHK_CRIOBJ_AT_EXPORT'  and ( VALUE = 'W' or VALUE = 'E' ) </compliant>
      <complianttext/>
      <noncompliant> NAME = 'CHK_CRIOBJ_AT_EXPORT'  and ( VALUE = 'W' or VALUE = 'E' ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
  <!-- END of Transport Parameters available as of FRUN 3.0 FP00 -->
  <configstore name="SAP_KERNEL">
    <checkitem desc="[p3-EXTENDED] Kernel release which note 2405918 recommends" id="CHANGE-A_g.3">
      <compliant>    NAME = 'KERN_REL' and 
                   ( ( substring(VALUE,0,3) &gt; '773' ) or
                     ( substring(VALUE,0,3) = '773' ) or
                     ( substring(VALUE,0,3) = '753') or
                     ( substring(VALUE,0,3) = '749') or
                     ( substring(VALUE,0,3) = '745') or
                     ( substring(VALUE,0,3) = '722') or
                     ( substring(VALUE,0,3) = '721') )
      </compliant>
      <noncompliant> NAME = 'KERN_REL' and not
                   ( ( substring(VALUE,0,3) &gt; '773' ) or
                     ( substring(VALUE,0,3) = '773' ) or
                     ( substring(VALUE,0,3) = '753') or
                     ( substring(VALUE,0,3) = '749') or
                     ( substring(VALUE,0,3) = '745') or
                     ( substring(VALUE,0,3) = '722') or
                     ( substring(VALUE,0,3) = '721') )
      </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Kernel patchlevel which note 2671160 recommends" id="CHANGE-A_g.3">
      <compliant>    NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &gt;= '1112'</compliant>
      <noncompliant> NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &lt; '1112'</noncompliant>
      <joinstore name="SAP_KERNEL" no_data_found="Yes">
        <joincompliant>      NAME = 'KERN_REL' and VALUE = '721_REL'</joincompliant>
        <joinnoncompliant>   NAME = 'KERN_REL' and VALUE = '721_REL'</joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Kernel patchlevel which note 2671160 recommends" id="CHANGE-A_g.3">
      <compliant>    NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &gt;= '0625'</compliant>
      <noncompliant> NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &lt; '0625'</noncompliant>
      <joinstore name="SAP_KERNEL" no_data_found="Yes">
        <joincompliant>      NAME = 'KERN_REL' and VALUE = '722_REL'</joincompliant>
        <joinnoncompliant>   NAME = 'KERN_REL' and VALUE = '722_REL'</joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Kernel patchlevel which note 2671160 recommends" id="CHANGE-A_g.3">
      <compliant>    NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &gt;= '0810'</compliant>
      <noncompliant> NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &lt; '0810'</noncompliant>
      <joinstore name="SAP_KERNEL" no_data_found="Yes">
        <joincompliant>      NAME = 'KERN_REL' and VALUE = '745_REL'</joincompliant>
        <joinnoncompliant>   NAME = 'KERN_REL' and VALUE = '745_REL'</joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Kernel patchlevel which note 2671160 recommends" id="CHANGE-A_g.3">
      <compliant>    NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &gt;= '0521'</compliant>
      <noncompliant> NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &lt; '0521'</noncompliant>
      <joinstore name="SAP_KERNEL" no_data_found="Yes">
        <joincompliant>      NAME = 'KERN_REL' and VALUE = '749_REL'</joincompliant>
        <joinnoncompliant>   NAME = 'KERN_REL' and VALUE = '749_REL'</joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Kernel patchlevel which note 2671160 recommends" id="CHANGE-A_g.3">
      <compliant>    NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &gt;= '0220'</compliant>
      <noncompliant> NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &lt; '0220'</noncompliant>
      <joinstore name="SAP_KERNEL" no_data_found="Yes">
        <joincompliant>      NAME = 'KERN_REL' and VALUE = '753_REL'</joincompliant>
        <joinnoncompliant>   NAME = 'KERN_REL' and VALUE = '753_REL'</joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Kernel patchlevel which note 2671160 recommends" id="CHANGE-A_g.3">
      <compliant>    NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &gt;= '0011'</compliant>
      <noncompliant> NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &lt; '0011'</noncompliant>
      <joinstore name="SAP_KERNEL" no_data_found="Yes">
        <joincompliant>      NAME = 'KERN_REL' and VALUE = '773_REL'</joincompliant>
        <joinnoncompliant>   NAME = 'KERN_REL' and VALUE = '773_REL'</joinnoncompliant>
      </joinstore>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Kernel patchlevel which note 2671160 recommends" id="CHANGE-A_g.3">
      <compliant>    NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &gt;= '0000'</compliant>
      <noncompliant> NAME = 'KERN_PATCHLEVEL' and lpad(VALUE,4,'0') &lt; '0000'</noncompliant>
      <joinstore name="SAP_KERNEL" no_data_found="Yes">
        <joincompliant>      NAME = 'KERN_REL' and VALUE = '774_REL'</joincompliant>
        <joinnoncompliant>   NAME = 'KERN_REL' and VALUE = '774_REL'</joinnoncompliant>
      </joinstore>
    </checkitem>
  </configstore>
</targetsystem>