<?xml version="1.0" encoding="utf-8"?>
<!-- 
This Policy requires creation of a customizable store. 
sci_id may be different in your system and may require an adjustment. 
Configstore name is up to your choice and also may require adjustment.
Store definition can be performed based on information from the Configuration Validation Template. 

Reference:   SAP Security Baseline Template v2.4 (2022/12)
Link:        https://support.sap.com/content/dam/support/en_us/library/ssp/offerings-and-programs/support-services/sap-security-optimization-services-portfolio/Security_Baseline_Template_V2.zip
Version:     001
Date:        2023-04-28
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL2.4] Directory Traversal Protection (ABAP)" id="3AFILE_CSTO" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
 <configstore name="AUTH_COMB_USER" sci_id="">
    <checkitem desc="[p3-EXTENDED] Users with S_DATASET_STAR authorization" id="FILE-A_c.1" not_found="positive">
      <compliant>         COMB_ID = 'S_DATASET_STAR' and
                                    USER like '%' and
                                    ( RC =  'NONE'  or RC = 'OBJECT_NOT_EXISTING' or
                                      LOCKED = 'X' or
                                      INVALID = 'X' )
      </compliant>
      <complianttext/>
      <noncompliant>   COMB_ID = 'S_DATASET_STAR' and
                                    USER like '%' and not
                                    ( RC =  'NONE' or RC = 'OBJECT_NOT_EXISTING' or
                                      LOCKED = 'X' or
                                      INVALID = 'X' )
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Users with S_PATH_STAR authorization" id="FILE-A_b.1" not_found="positive">
      <compliant>         COMB_ID = 'S_PATH_STAR' and
                                    USER like '%' and
                                    ( RC =  'NONE'  or RC = 'OBJECT_NOT_EXISTING' or
                                      LOCKED = 'X' or
                                      INVALID = 'X' )
      </compliant>
      <complianttext/>
      <noncompliant>   COMB_ID = 'S_PATH_STAR' and
                                    USER like '%' and not
                                    ( RC =  'NONE' or RC = 'OBJECT_NOT_EXISTING' or
                                      LOCKED = 'X' or
                                      INVALID = 'X' )
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
  <configstore name="AUTH_COMB_ROLE" sci_id="">
    <checkitem desc="[p3-EXTENDED] Roles with S_DATASET_STAR authorization" id="FILE-A_c.2" not_found="positive">
      <compliant>         COMB_ID = 'S_DATASET_STAR' and
                                    ROLE like '%' and
                                    ROLE_TYPE like '%' and
                                    ( RC =  'NONE'  or RC = 'OBJECT_NOT_EXISTING' )
      </compliant>
      <complianttext/>
      <noncompliant>   COMB_ID = 'S_DATASET_STAR' and
                                    ROLE like '%' and
                                    ROLE_TYPE like '%' and not
                                    ( RC =  'NONE' or RC = 'OBJECT_NOT_EXISTING'  )
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Roles with S_PATH_STAR authorization" id="FILE-A_b.2" not_found="positive">
      <compliant>         COMB_ID = 'S_PATH_STAR' and
                                    ROLE like '%' and
                                    ROLE_TYPE like '%' and 
                                    ( RC =  'NONE'  or RC = 'OBJECT_NOT_EXISTING' )
      </compliant>
      <complianttext/>
      <noncompliant>   COMB_ID = 'S_PATH_STAR' and
                                    ROLE like '%' and
                                    ROLE_TYPE like '%' and not
                                    ( RC =  'NONE' or RC = 'OBJECT_NOT_EXISTING' )
      </noncompliant>
      <noncomplianttext/>
    </checkitem>
  </configstore>
</targetsystem>