<?xml version="1.0" encoding="utf-8"?>
<!-- 
Reference:   SAP Security Baseline Template v2.4 (2022/12)
Link:        https://support.sap.com/content/dam/support/en_us/library/ssp/offerings-and-programs/support-services/sap-security-optimization-services-portfolio/Security_Baseline_Template_V2.zip
Version:     001
Date:        2023-04-28
-->
<targetsystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" desc="[BL2.4] Password Policy (ABAP)" id="3APWDPOL" multisql="Yes" version="0000" xsi:schemaLocation="csa_policy.xsd">
  <!-- PWDPOL-A: Password Policy â€“ ABAP -->
  <configstore name="ABAP_INSTANCE_PAHI">
    <checkitem desc="[p3-EXTENDED] Maximum password expiration time" id="PWDPOL-A_c.1">
      <compliant>       NAME = 'login/password_expiration_time' and lpad(VALUE,4,'0') between '0001' and '0183' </compliant>
      <noncompliant>    NAME = 'login/password_expiration_time' and NOT lpad(VALUE,4,'0')between '0001' and '0183' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (digits) must exist (profile parameter)" id="PWDPOL-A_i.1.1">
      <compliant>          NAME = 'login/min_password_digits' and lpad(VALUE,4,'0') &gt;= '0001' </compliant>
      <noncompliant>    NAME = 'login/min_password_digits' and not lpad(VALUE,4,'0') &gt;= '0001'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (letters) must exist (profile parameter)" id="PWDPOL-A_i.2.1">
      <compliant>          NAME = 'login/min_password_letters' and lpad(VALUE,4,'0') &gt;= '0001'  </compliant>
      <noncompliant>    NAME = 'login/min_password_letters' and not lpad(VALUE,4,'0') &gt;= '0001' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (lowercase letters) must exist (profile parameter)" id="PWDPOL-A_i.3.1">
      <compliant>          NAME = 'login/min_password_lowercase' and lpad(VALUE,4,'0') &gt;= '0001'  </compliant>
      <noncompliant>    NAME = 'login/min_password_lowercase' and not lpad(VALUE,4,'0') &gt;= '0001'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (uppercase letters) must exist (profile parameter)" id="PWDPOL-A_i.4.1">
      <compliant>          NAME = 'login/min_password_uppercase' and lpad(VALUE,4,'0') &gt;= '0001' </compliant>
      <noncompliant>   NAME = 'login/min_password_uppercase' and not lpad(VALUE,4,'0') &gt;= '0001'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (special characters) must exist (profile parameter)" id="PWDPOL-A_i.5.1">
      <compliant>          NAME = 'login/min_password_specials' and lpad(VALUE,4,'0') &gt;= '0001' </compliant>
      <noncompliant>   NAME = 'login/min_password_specials' and not lpad(VALUE,4,'0') &gt;= '0001'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for number of different previous passwords must exist (profile parameter)" id="PWDPOL-A_i.6.1">
      <compliant>          NAME = 'login/min_password_diff' and lpad(VALUE,4,'0') &gt;= '0003'  </compliant>
      <noncompliant>    NAME = 'login/min_password_diff' and not lpad(VALUE,4,'0') &gt;= '0003' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for disabling password logon must exist (profile parameter)" id="PWDPOL-A_j.1.1">
      <compliant>          NAME = 'login/disable_password_logon' and length(VALUE) &gt;= '1'  </compliant>
      <noncompliant>    NAME = 'login/disable_password_logon' and not length(VALUE) &gt;= '1' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for maximum failed logon attempts must exist (profile parameter)" id="PWDPOL-A_j.3.1">
      <compliant>          NAME = 'login/fails_to_user_lock' and lpad(VALUE,4,'0') between '0001' and '0005' </compliant>
      <noncompliant>    NAME = 'login/fails_to_user_lock' and not lpad(VALUE,4,'0') between '0001' and '0005' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for automatic user unlock (after failed logins) must exist (profile parameter)" id="PWDPOL-A_j.4.1">
      <compliant>          NAME = 'login/failed_user_auto_unlock' and VALUE = '0' </compliant>
      <noncompliant>    NAME = 'login/failed_user_auto_unlock' and VALUE != '0' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for maximum idle time of productive passwords must exist (profile parameter)" id="PWDPOL-A_j.5.1">
      <compliant>          NAME = 'login/password_max_idle_productive' and lpad(VALUE,4,'0') between '0001' and '0180' </compliant>
      <noncompliant>    NAME = 'login/password_max_idle_productive' and not lpad(VALUE,4,'0') between '0001' and '0180'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password change for SSO must exist (profile parameter)" id="PWDPOL-A_j.6.1">
      <compliant>          NAME = 'login/password_change_for_SSO' and VALUE = '1' </compliant>
      <noncompliant>    NAME = 'login/password_change_for_SSO' and VALUE != '1' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password history size must exist (profile parameter)" id="PWDPOL-A_j.7.1">
      <compliant>          NAME = 'login/password_history_size' and lpad(VALUE,4,'0') &gt;= '0005' </compliant>
      <noncompliant>    NAME = 'login/password_history_size' and not lpad(VALUE,4,'0') &gt;= '0005' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Password hash algorithm" id="PWDPOL-A_k">
      <compliant>   NAME = 'login/password_hash_algorithm' and 
                    VALUE like_regexpr 'encoding=RFC2307, ?algorithm=(iSSHA-256|iSSHA-384|iSSHA-512), ?iterations=(\d{5,}), ?saltsize=(128|256)'
      </compliant>
      <noncompliant>NAME = 'login/password_hash_algorithm' and not 
                    VALUE like_regexpr 'encoding=RFC2307, ?algorithm=(iSSHA-256|iSSHA-384|iSSHA-512), ?iterations=(\d{5,}), ?saltsize=(128|256)'
      </noncompliant>
    </checkitem>
    <!-- Alternative Syntax supporting different order of entries
    <checkitem desc="Configure a compliant password policy (login/password_hash_algorithm)" id="PWDPOL-A_k">
      <compliant>NAME = 'login/password_hash_algorithm' and 
                 VALUE like_regexpr 'encoding=RFC2307'  and
                ( VALUE like_regexpr 'algorithm=iSSHA-256' OR VALUE like_regexpr 'algorithm=iSSHA-384' OR VALUE like_regexpr 'algorithm=iSSHA-512') and
                ( VALUE like_regexpr 'saltsize=128' OR VALUE like_regexpr 'saltsize=256' )
      </compliant>
      <noncompliant>NAME = 'login/password_hash_algorithm' and 
               ( ( not VALUE like_regexpr 'encoding=RFC2307' ) OR
                ( not VALUE like_regexpr 'algorithm=iSSHA-256' AND not VALUE like_regexpr 'algorithm=iSSHA-384' AND not VALUE like_regexpr 'algorithm=iSSHA-512') OR
                ( not VALUE like_regexpr 'saltsize=128' AND not VALUE like_regexpr 'saltsize=256' ) )
      </noncompliant>
    </checkitem>
-->
  </configstore>
  <configstore name="AUTH_SECURITY_POLICY">
    <checkitem desc="[p3-EXTENDED] Rule for password change attribute is noncompliant (security policy attribute)" id="PWDPOL-A_c.2" not_found="positive">
      <compliant>       ATTRIB_KEY = 'PASSWORD_CHANGE_INTERVAL' and NAME like '%' and CLIENT != '[XXX]' and lpad(ATTRIB_VALUE,4,'0') between '0001' and '0183' </compliant>
      <noncompliant>    ATTRIB_KEY = 'PASSWORD_CHANGE_INTERVAL' and NAME like '%' and CLIENT != '[XXX]'  and NOT lpad(ATTRIB_VALUE,4,'0') between '0001' and '0183' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (digits) is noncompliant (security policy attribute)" id="PWDPOL-A_i.1.2" not_found="positive">
      <compliant>          ATTRIB_KEY = 'MIN_PASSWORD_DIGITS' and NAME like '%'  and  ( CLIENT != '[XXX]' and lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001' ) </compliant>
      <noncompliant>    ATTRIB_KEY = 'MIN_PASSWORD_DIGITS' and NAME like '%' and CLIENT != '[XXX]'  and NOT lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (letters) is noncompliant (security policy attribute)" id="PWDPOL-A_i.2.2" not_found="positive">
      <compliant>          ATTRIB_KEY = 'MIN_PASSWORD_LETTERS' and NAME like '%'  and  ( CLIENT != '[XXX]' and lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001' ) </compliant>
      <noncompliant>    ATTRIB_KEY = 'MIN_PASSWORD_LETTERS' and NAME like '%' and CLIENT != '[XXX]'  and NOT lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (lowercase letters) is noncompliant (security policy attribute)" id="PWDPOL-A_i.3.2" not_found="positive">
      <compliant>          ATTRIB_KEY = 'MIN_PASSWORD_LOWERCASE' and NAME like '%'  and  ( CLIENT != '[XXX]' and lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001' ) </compliant>
      <noncompliant>    ATTRIB_KEY = 'MIN_PASSWORD_LOWERCASE' and NAME like '%' and CLIENT != '[XXX]'  and NOT lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (uppercase letters) is noncompliant (security policy attribute)" id="PWDPOL-A_i.4.2" not_found="positive">
      <compliant>         ATTRIB_KEY = 'MIN_PASSWORD_UPPERCASE' and NAME like '%'  and  ( CLIENT != '[XXX]' and lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001' ) </compliant>
      <noncompliant>   ATTRIB_KEY = 'MIN_PASSWORD_UPPERCASE' and NAME like '%' and CLIENT != '[XXX]'  and NOT lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password complexity (special characters) is noncompliant (security policy attribute)" id="PWDPOL-A_i.5.2" not_found="positive">
      <compliant>          ATTRIB_KEY = 'MIN_PASSWORD_SPECIALS' and NAME like '%'  and  ( CLIENT != '[XXX]' and lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001' ) </compliant>
      <noncompliant>   ATTRIB_KEY = 'MIN_PASSWORD_SPECIALS' and NAME like '%' and CLIENT != '[XXX]'  and NOT lpad(ATTRIB_VALUE,4,'0')  &gt;= '0001'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for number of different previous passwords is noncompliant(security policy attribute)" id="PWDPOL-A_i.6.2" not_found="positive">
      <compliant>          ATTRIB_KEY = 'MIN_PASSWORD_DIFFERENCE' and NAME like '%'  and  ( CLIENT != '[XXX]' and lpad(ATTRIB_VALUE,4,'0')  &gt;= '0003' ) </compliant>
      <noncompliant>   ATTRIB_KEY = 'MIN_PASSWORD_DIFFERENCE' and NAME like '%' and CLIENT != '[XXX]'  and NOT lpad(ATTRIB_VALUE,4,'0')  &gt;= '0003' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for disabling password logon is noncompliant (security policy attribute)" id="PWDPOL-A_j.1.2" not_found="positive">
      <compliant>         ATTRIB_KEY = 'DISABLE_PASSWORD_LOGON' and NAME like '%' and ( CLIENT != '[XXX]' and length(ATTRIB_VALUE) &gt;= '1'  ) </compliant>
      <noncompliant>  ATTRIB_KEY = 'DISABLE_PASSWORD_LOGON' and NAME like '%' and CLIENT != '[XXX]'  and NOT length(ATTRIB_VALUE) &gt;= '1' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for disabling ticket logon is noncompliant (security policy attribute)" id="PWDPOL-A_j.2.2" not_found="positive">
      <compliant>         ATTRIB_KEY = 'DISABLE_TICKET_LOGON' and NAME like '%' and ( CLIENT != '[XXX]' and length(ATTRIB_VALUE) &gt;= '1'  ) </compliant>
      <noncompliant>  ATTRIB_KEY = 'DISABLE_TICKET_LOGON' and NAME like '%' and CLIENT != '[XXX]'  and NOT length(ATTRIB_VALUE) &gt;= '1' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for maximum failed logon attempts is noncompliant (security policy attribute)" id="PWDPOL-A_j.3.2" not_found="positive">
      <compliant>          ATTRIB_KEY = 'MAX_FAILED_PASSWORD_LOGON_ATTEMPTS' and NAME like '%'  and  ( CLIENT != '[XXX]' and lpad(ATTRIB_VALUE,4,'0')  &lt;= '0005' ) </compliant>
      <noncompliant>   ATTRIB_KEY = 'MAX_FAILED_PASSWORD_LOGON_ATTEMPTS' and NAME like '%' and CLIENT != '[XXX]'  and NOT lpad(ATTRIB_VALUE,4,'0')  &lt;= '0005' </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for automatic user unlock (after failed logins) is noncompliant (security policy attribute)" id="PWDPOL-A_j.4.2" not_found="positive">
      <compliant>         ATTRIB_KEY = 'PASSWORD_LOCK_EXPIRATION' and NAME like '%' and   ( CLIENT != '[XXX]' and  ATTRIB_VALUE = '0'  ) </compliant>
      <noncompliant>  ATTRIB_KEY = 'PASSWORD_LOCK_EXPIRATION' and NAME like '%' and CLIENT != '[XXX]'  and  ATTRIB_VALUE = '0'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for maximum idle time of productive passwords is noncompliant(security policy attribute)" id="PWDPOL-A_j.5.2" not_found="positive">
      <compliant>          ATTRIB_KEY = 'MAX_PASSWORD_IDLE_PRODUCTIVE' and NAME like '%' and   ( CLIENT != '[XXX]' and  lpad(ATTRIB_VALUE,4,'0') between '0001' and '0180'  ) </compliant>
      <noncompliant>   ATTRIB_KEY = 'MAX_PASSWORD_IDLE_PRODUCTIVE' and NAME like '%' and CLIENT != '[XXX]'  and  NOT lpad(ATTRIB_VALUE,4,'0') between '0001' and '0180'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password change wait time is noncompliant (security policy attribute)" id="PWDPOL-A_j.6.2" not_found="positive">
      <compliant>         ATTRIB_KEY = 'MIN_PASSWORD_CHANGE_WAITTIME' and NAME like '%' and ( CLIENT != '[XXX]' and length(ATTRIB_VALUE) &gt; '0'  ) </compliant>
      <noncompliant>  ATTRIB_KEY = 'MIN_PASSWORD_CHANGE_WAITTIME' and NAME like '%' and CLIENT != '[XXX]'  and NOT length(ATTRIB_VALUE) &gt; '0'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password change for SSO is noncompliant (security policy attribute)" id="PWDPOL-A_j.7.2" not_found="positive">
      <compliant>         ATTRIB_KEY = 'PASSWORD_CHANGE_FOR_SSO' and NAME like '%' and   ( CLIENT != '[XXX]' and ATTRIB_VALUE = '1'  ) </compliant>
      <noncompliant>   ATTRIB_KEY = 'PASSWORD_CHANGE_FOR_SSO' and NAME like '%' and CLIENT != '[XXX]'  and ATTRIB_VALUE !=  '1'  </noncompliant>
    </checkitem>
    <checkitem desc="[p3-EXTENDED] Rule for password history size is noncompliant (security policy attribute)" id="PWDPOL-A_j.8.2" not_found="positive">
      <compliant>         ATTRIB_KEY = 'PASSWORD_HISTORY_SIZE' and NAME like '%' and   ( CLIENT != '[XXX]' and  lpad(ATTRIB_VALUE,4,'0')  &gt;= '0005') </compliant>
      <noncompliant>  ATTRIB_KEY = 'PASSWORD_HISTORY_SIZE' and NAME like '%' and CLIENT != '[XXX]'  and NOT lpad(ATTRIB_VALUE,4,'0')  &gt;= '0005'  </noncompliant>
    </checkitem>
  </configstore>
</targetsystem>